File: Python-Library-Project/periodic.py
#Sumeet Tulsani
from media import Media

class Periodic(Media):
    def __init__(self,call_number,title,subjects,author,description,publisher,publishing_history,series,notes,related_titles,other_forms_of_title,gov_doc_number):
        Media.__init__(self,call_number,title,subjects,notes)
        self.author = author
        self.description = description
        self.publisher = publisher
        self.publishing_history = publishing_history
        self.series = series
        self.related_titles = related_titles
        self.other_forms_of_title = other_forms_of_title
        self.gov_doc_number = gov_doc_number

    def display(self):
        print ("---------------PERIODIC-----------------");
        print ("CALL NUMBER::",self.call_number);
        print ("TITLE::",self.title);
        print ("SUBJECTS::",self.subjects);
        print ("AUTHOR::",self.author);
        print ("DESCRIPTION::",self.description);
        print ("PUBLISHER::",self.publisher);
        print ("PUBLISHING_HISTORY::",self.publishing_history);
        print ("SERIES::",self.series);
        print ("NOTES::",self.notes);
        print ("RELATED_TITLES::",self.related_titles);
        print ("OTHER_FORMS_OF_TITLE::",self.other_forms_of_title);
        print ("GOV_DOC_NUMBER::",self.gov_doc_number);

    def compare_other(self,searchby):
        if (self.description.find(searchby) != -1) or  (self.notes.find(searchby) != -1) or (self.series.find(searchby) != -1) or (self.related_titles.find(searchby) != -1):
            return True
        else:
            return False

File: LibraryProject/periodic.cpp
#include<iostream>
#include<string>
#include<vector>
#include"periodic.h"
#include "boost/algorithm/string.hpp"

using namespace std;
using namespace boost;

Periodic::Periodic(const string& call_number,const string& title,const string& subjects,
                   const string& author,const string& description,const string& publisher,
                   const string& publishing_history,const string& series,const string& notes,
                   const string& related_titles,const string& other_forms_of_title,
                   const string& gov_doc_number): Media(call_number,title,subjects,notes),
                   author(author),description(description),publisher(publisher),
                   publishing_history(publishing_history),series(series),related_titles(related_titles),
                   other_forms_of_title(other_forms_of_title),gov_doc_number(gov_doc_number)
{

}

void Periodic::display() const
{
    cout << "-----------PERIODIC----------" << endl;
    cout << "CALL NUMBER::" << call_number << endl << "TITLE::" << title << endl
    << "SUBJECTS::" << subjects << endl<< "AUTHOR::" << author << endl << "DESCRIPTION::" <<
    description << endl << "PUBLISHER::" << publisher << endl <<  "PUBLISHING HISTORY::" <<
    publishing_history << endl << "SERIES::" << series << endl <<  "NOTES::"
     << notes << endl << "RELATED TITLES::" << related_titles << endl << "OTHER FORMS OF TITLE::"
     << other_forms_of_title << endl << "GOV DOC NUMBER::" << gov_doc_number << endl;
}

bool Periodic::compare_other(const string& searchby) const
{
    size_t found = description.find(searchby);
    size_t found1 = notes.find(searchby);
    size_t found2 = series.find(searchby);
    size_t found3 = related_titles.find(searchby);
    if((found!=string::npos) || (found1!=string::npos) || (found2!=string::npos) || (found3!=string::npos))
    {
        return true;
    }
    else
    {
        return false;
    }
}

